cmd/ops_agent_uap_plugin/service_linux.go

```diff
@@ -95,41 +95,61 @@ func (ps *OpsAgentPluginServer) Start(ctx context.Context, msg *pb.StartRequest)
 	// Find existing ops agent installation, and conflicting legacy agent installation.
 	foundConflictingInstallations, err := findPreExistentAgents(pContext, ps.runCommand, AgentSystemdServiceNames)
 	if foundConflictingInstallations || err != nil {
-		ps.Stop(ctx, &pb.StopRequest{Cleanup: false})
-		log.Printf("Start() failed: %s", err)
+		ps.logAndSetPluginError(fmt.Sprintf("Start() failed, because it detected agent installations unmanaged by the VM Extension Manager: %s", err), false)
+		cancel()
 		return &pb.StartResponse{}, nil
 	}
 
 	// Receive config from the Start request and write it to the Ops Agent config file.
 	if err := writeCustomConfigToFile(msg, OpsAgentConfigLocationLinux); err != nil {
-		log.Printf("Start() failed: %s", err)
-		ps.Stop(ctx, &pb.StopRequest{Cleanup: false})
-		log.Printf("Start() failed to write the custom Ops Agent config to file: %s", err)
+		ps.logAndSetPluginError(fmt.Sprintf("Start() failed to write the custom Ops Agent config to file: %s", err), false)
+		cancel()
 		return &pb.StartResponse{}, nil
 	}
 
 	// Ops Agent config validation
 	if err := validateOpsAgentConfig(pContext, pluginInstallDir, pluginStateDir, ps.runCommand); err != nil {
-		log.Printf("Start() failed: %s", err)
-		ps.Stop(ctx, &pb.StopRequest{Cleanup: false})
-		log.Printf("Start() failed to validate the Ops Agent config: %s", err)
+		ps.logAndSetPluginError(fmt.Sprintf("Start() failed to validate the Ops Agent config: %s", err), false)
+		cancel()
 		return &pb.StartResponse{}, nil
 	}
 	// Subagent config generation
 	if err := generateSubagentConfigs(pContext, ps.runCommand, pluginInstallDir, pluginStateDir); err != nil {
-		log.Printf("Start() failed: %s", err)
-		ps.Stop(ctx, &pb.StopRequest{Cleanup: false})
-		log.Printf("Start() failed to generate subagent configs: %s", err)
+		ps.logAndSetPluginError(fmt.Sprintf("Start() failed to generate subagent configs: %s", err), false)
+		cancel()
 		return &pb.StartResponse{}, nil
 	}
 
+	ps.mu.Lock()
+	defer ps.mu.Unlock()
+	ps.cancel = cancel
+
 	// the subagent startups
-	cancelFunc := func() {
-		ps.Stop(ctx, &pb.StopRequest{Cleanup: false})
-	}
-	go runSubagents(pContext, cancelFunc, pluginInstallDir, pluginStateDir, runSubAgentCommand, ps.runCommand)
+	go runSubagents(pContext, ps.cancelAndSetPluginError, pluginInstallDir, pluginStateDir, runSubAgentCommand, ps.runCommand)
 	return &pb.StartResponse{}, nil
 }
+func (ps *OpsAgentPluginServer) cancelAndSetPluginError(e *OpsAgentPluginError) {
+	ps.mu.Lock()
+	defer ps.mu.Unlock()
+	if ps.cancel != nil {
+		ps.cancel()
+		ps.cancel = nil
+	}
+	if e != nil {
+		ps.pluginError = e
+	}
+}
+
+func (ps *OpsAgentPluginServer) logAndSetPluginError(errMsg string, isFatal bool) {
+	ps.mu.Lock()
+	defer ps.mu.Unlock()
+
+	ps.pluginError = &OpsAgentPluginError{
+		Message: errMsg,
+		IsFatal: isFatal,
```

Comment:
It's not immediately clear from reading the code when `isFatal` should be true, especially critical errors that render the Ops Agent useless (e.g. bad config) are not considered "Fatal" by UAP standards.

Maybe we change this to something like `needsRestart` or `shouldRestart` to be more explicit?